Lint failures:

```
installing ruff...
F401 [*] `plextraktsync.config.PlexServerConfig.PlexServerConfig` imported but unused
  --> tests/test_config.py:9:51
   |
 8 | from plextraktsync.config.Config import Config
 9 | from plextraktsync.config.PlexServerConfig import PlexServerConfig
   |                                                   ^^^^^^^^^^^^^^^^
10 | from plextraktsync.config.ServerConfigFactory import ServerConfigFactory
11 | from plextraktsync.factory import factory
   |
help: Remove unused import: `plextraktsync.config.PlexServerConfig.PlexServerConfig`

Found 1 error.
[*] 1 fixable with the `--fix` option.
Error: Process completed with exit code 1.
```

--- a/plextraktsync/config/PlexServerConfig.py
+++ b/plextraktsync/config/PlexServerConfig.py
@@ -20,6 +20,12 @@ class PlexServerConfig:
         data = asdict(self)
         del data["name"]
 
+        # Remove optional fields that are None so they don't overwrite
+        # existing values when merging (e.g. preserving user config on re-login)
+        for key in ("id", "config"):
+            if data[key] is None:
+                del data[key]
+
         return data
 
     @property

--- a/tests/test_config.py
+++ b/tests/test_config.py
@@ -6,6 +6,7 @@ from os.path import join
 import pytest
 
 from plextraktsync.config.Config import Config
+from plextraktsync.config.ServerConfigFactory import ServerConfigFactory
 from plextraktsync.factory import factory
 
 
@@ -37,6 +38,35 @@ def test_sync_config():
     sync_config = Config(config_file).sync
 
     assert sync_config.plex_to_trakt["collection"] is False
+
+
+def test_relogin_preserves_server_config():
+    """Re-login must not overwrite the user's custom server config (libraries, etc.)."""
+    sc = ServerConfigFactory()
+    sc.loaded = True
+
+    # Simulate initial login with a custom config (libraries whitelist)
+    sc.add_server(
+        name="myserver",
+        token="old-token",
+        urls=["https://plex.example.com:32400"],
+        id="abc123",
+        config={"libraries": ["Movies", "Series"]},
+    )
+    assert sc.servers["myserver"]["config"] == {"libraries": ["Movies", "Series"]}
+
+    # Simulate re-login: only token, urls, id are passed — config is not supplied
+    sc.add_server(
+        name="myserver",
+        token="new-token",
+        urls=["https://plex.example.com:32400"],
+        id="abc123",
+    )
+
+    # Token should be updated
+    assert sc.servers["myserver"]["token"] == "new-token"
+    # Config must be preserved, not overwritten with None
+    assert sc.servers["myserver"]["config"] == {"libraries": ["Movies", "Series"]}
 
 
 def test_http_config():

